pkg/cloud/aws/services/ec2/ami.go

```diff
@@ -80,8 +85,43 @@ func (s *Service) defaultAMILookup(baseOS, baseOSVersion, kubernetesVersion stri
 	if len(out.Images) == 0 {
 		return "", errors.Errorf("found no AMIs with the name: %q", amiName(baseOS, baseOSVersion, kubernetesVersion))
 	}
-	klog.V(2).Infof("Using AMI: %q", aws.StringValue(out.Images[0].ImageId))
-	return aws.StringValue(out.Images[0].ImageId), nil
+	latestImage := getLatestImage(out.Images)
+	klog.V(2).Infof("Using AMI: %q", aws.StringValue(latestImage.ImageId))
+	return aws.StringValue(latestImage.ImageId), nil
+}
+
+type images []*ec2.Image
+
+// Len is the number of elements in the collection.
+func (i images) Len() int {
+	return len(i)
+}
+
+// Less reports whether the element with
+// index i should sort before the element with index j.
+func (i images) Less(k, j int) bool {
+	firstTime, err := time.Parse(createDateTimestampFormat, aws.StringValue(i[k].CreationDate))
+	if err != nil {
+		klog.Infof("unable to parse an AMI creation timestamp: %q", aws.StringValue(i[k].CreationDate))
+		return false
+	}
+	secondTime, err := time.Parse(createDateTimestampFormat, aws.StringValue(i[j].CreationDate))
+	if err != nil {
+		klog.Infof("unable to parse an AMI creation timestamp: %q", aws.StringValue(i[j].CreationDate))
+		return false
+	}
+	return firstTime.Before(secondTime)
+}
+
+// Swap swaps the elements with indexes i and j.
+func (i images) Swap(k, j int) {
+	i[k], i[j] = i[j], i[k]
+}
+
+func getLatestImage(imgs []*ec2.Image) *ec2.Image {
+	// old to new (newest one is last)
+	sort.Sort(images(imgs))
+	return imgs[len(imgs)-1]
```

Comment:
This can panic if imgs is empty

Reply:
true, but the assumption of this code, which perhaps I should document, is that it won't be empty

https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/563/files/6715f7002a93f2b3c590abd5d761e2f83d98eb7b#diff-26464ba5826fa06de3ee8d0a402f6efcR85